analyzers/src/SonarAnalyzer.CSharp/Rules/GenericTypeParametersRequired.cs

```diff
@@ -31,35 +31,50 @@ public sealed class GenericTypeParametersRequired : SonarDiagnosticAnalyzer
 
         protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var methodDeclaration = (MethodDeclarationSyntax)c.Node;
+
+                var typeParameters = methodDeclaration
+                        .TypeParameterList
+                        ?.Parameters
+                        .Select(p => c.SemanticModel.GetDeclaredSymbol(p));
+
+                if (typeParameters == null)
                 {
-                    var methodDeclaration = (MethodDeclarationSyntax)c.Node;
+                    return;
+                }
 
-                    var typeParameters = methodDeclaration
-                            .TypeParameterList
-                            ?.Parameters
-                            .Select(p => c.SemanticModel.GetDeclaredSymbol(p));
+                var argumentTypes = methodDeclaration
+                        .ParameterList
+                        .Parameters
+                        .Select(p => c.SemanticModel.GetDeclaredSymbol(p)?.Type);
 
-                    if (typeParameters == null)
-                    {
-                        return;
-                    }
+                var constraintTypes = methodDeclaration
+                    .ConstraintClauses
+                    .OfType<TypeParameterConstraintClauseSyntax>()
+                    .SelectMany(p => p
+                                    .Constraints
+                                    .OfType<TypeConstraintSyntax>()
+                                    .Select(t => c.SemanticModel.GetSymbolInfo(t.Type).Symbol))
```

Comment:
AFAIR the result of GetSymbolInfo be null (for edge cases like being in the IDE and writing code and the code being incomplete when the rule is executed; and I think there's other cases as well)

can you use an existing extension method like `GetSymbolOrCandidateSymbol` for this?